src/graphql-aspnet/Middleware/FieldExecution/Components/ProcessChildFieldsMiddleware.cs

```diff
@@ -193,6 +195,40 @@ private async Task ProcessDownStreamFieldContexts(GraphFieldExecutionContext con
             }
         }
 
+        /// <summary>
+        /// Inspects both the childcontext and the invoked task to determine
+        /// if any messages were found or any unhandled exceptions were thrown then
+        /// captures the results into the parent context for upstream processing or
+        /// reporting.
+        /// </summary>
+        /// <param name="parentContext">The parent context that invoked the child.</param>
+        /// <param name="childContext">The child context that was invoked.</param>
+        /// <param name="childExecutionTask">The actual task representing the child
+        /// pipeline.</param>
+        private void CaptureChildFieldExecutionResults(
+            GraphFieldExecutionContext parentContext,
+            GraphFieldExecutionContext childContext,
+            Task childExecutionTask)
+        {
+            // capture any messages that the child context may have internally reported
+            parentContext.Messages.AddRange(childContext.Messages);
+
+            // inspect the task for faulting and capture any exceptions as critical errors
+            if (childExecutionTask.IsFaulted)
+            {
+                var exception = childExecutionTask.UnwrapException();
+                exception = exception ?? new Exception(
+                    "Unknown Error. The child field execution pipeline indicated a fault but did not " +
```

Comment:
the general exception case should probably indicate which child failed as it may be the only message to surface in some cases.